packages/trader/src/Modules/Trading/Components/Form/ContractType/ContractTypeInfo/contract-type-info.jsx

```diff
@@ -29,8 +30,32 @@ const Info = ({ handleNavigationClick, handleSelect, initial_index, item, list }
     const handleItemSelect = active_index => {
         setCarouselIndex(contract_types[active_index].value);
         handleNavigationClick(contract_types[active_index]);
+
+        RudderStack.track('ce_trade_types_form', {
```

Comment:
@akmal-deriv Just a suggestion, In my opinion, all the events related to `ce_trade_types_form` could be abstracted into a hook to avoid the complexity of the components, What do you think? 🤔

Reply:
Imo we don't need a hook anymore for this, `Rudderstack.track` is already an abstraction of `ce_trade_types_form`, it defines types for `ce_trade_types_form` 😅

Reply:
There's also a lot of hooks already in `packages/hooks`, that would mean creating a new hook for every single Rudderstack event 😓

Reply:
Agree with the second point as well, if we make a separate hook for each event that would bloat the codebase, if we make it generalised that is already what we have in @deriv/analytics

Reply:
It was just a suggestion, Please feel free to ignore this comment 🌹🙇🏻 

It's not just about reusability, It's also about maintainability and separation of concern, I would recommend having a look at this article I sent [here](https://deriv-group.slack.com/archives/C3AHMSM3P/p1689128532619929) 🙇🏻